### PR TITLE
fix(951240): update rule based on regex

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

This is the last fix that will enable the regexp updated lint check (#2744).

### CODE REVIEW

**Review by theseion — COMMENTED** [review]
Looks good to me, but what's with the regressions?

**Comment by fzipi** [comment]
Huh. This is weird. I cannot see a data file for this rule 🤔

**Comment by theseion** [comment]
There isn't... We added one for 951230 but the rule was changed in 286092cca873a8521b21ad4431704cb1d63cd8a3, so it's no longer chained (need to rename the data file).

Could you also update the comment on 951230 to match the other regexp-assemble comments?

How did you make the change to 951240 if there's no data file...? 😅 Anyway, we should take the opportunity and create the data file.

**Comment by fzipi** [comment]
Decomposed regexp, and looks right. But the generated changes the rule. Probably because the initial `(?i` in the original rule, instead of using `(?i)(`?

File: util/regexp-assemble/data/951240.data [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+##!+ i
+
+PostgreSQL query failed:
+pg_query\(\) \[:
+pg_exec\(\) \[:
+PostgreSQL.{1,20}ERROR
+Warning.*\bpg_.*
+valid PostgreSQL result
+Npgsql\.
+PG::[a-zA-Z]*Error
```
Comment (theseion) [inline util/regexp-assemble/data/951240.data:13]:
```suggestion
PG::[a-z]*Error
```

File: rules/RESPONSE-951-DATA-LEAKAGES-SQL.conf [hunk 2]
```diff
@@ -320,7 +320,7 @@ SecRule RESPONSE_BODY "@rx (?i)(?:MyS(?:QL server version for the right syntax t
     setvar:'tx.outbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}',\
     setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}'"
 
-SecRule RESPONSE_BODY "@rx (?i:PostgreSQL query failed:|pg_query\(\) \[:|pg_exec\(\) \[:|PostgreSQL.{1,20}ERROR|Warning.*\bpg_.*|valid PostgreSQL result|Npgsql\.|PG::[a-zA-Z]*Error|Supplied argument is not a valid PostgreSQL .*? resource|Unable to connect to PostgreSQL server)" \
```
Comment (theseion) [inline rules/RESPONSE-951-DATA-LEAKAGES-SQL.conf:323]: Add regexp-assemble comment before rule.

Comment (theseion) [inline rules/RESPONSE-951-DATA-LEAKAGES-SQL.conf:323]: Fix regexp-assemble comment on 951230.

**Review by theseion — CHANGES_REQUESTED** [review]

**Comment by theseion** [comment]
> Decomposed regexp, and looks right. But the generated changes the rule. Probably because the initial `(?i` in the original rule, instead of using `(?i)(`?

Yes, flags in regexp-assemble are always prefixed as `(?<flag>)`, but `(?<flag>:<regex>)` is a valid form.

**Comment by theseion** [comment]
I wanted the comments to be the same as the others, so I fixed my own review comments too.

**Review by theseion — APPROVED** [review]

**Pull request merged**